adapters/YahooAdapter/YahooAdapter/GADMAdapterYahooBaseClass.m

```diff
@@ -0,0 +1,335 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "GADMAdapterYahooBaseClass.h"
+#import "GADMAdapterYahooConstants.h"
+#import "GADMAdapterYahooNativeAd.h"
+#import "GADMAdapterYahooUtils.h"
+
+@interface GADMAdapterYahooBaseClass () <YASInlineAdViewDelegate, YASInterstitialAdDelegate>
+@end
+
+@implementation GADMAdapterYahooBaseClass {
+  /// Yahoo native ad mapper.
+  GADMAdapterYahooNativeAd *_nativeAd;
+
+  /// Connector from the Google Mobile Ads SDK to receive ad configurations.
+  __weak id<GADMAdNetworkConnector> _connector;
+}
+
+#pragma mark - Logger
+
++ (nonnull YASLogger *)logger {
+  static YASLogger *_logger = nil;
+  static dispatch_once_t onceToken;
+  dispatch_once(&onceToken, ^{
+    _logger = [YASLogger loggerForClass:[GADMAdapterYahooBaseClass class]];
+  });
+  return _logger;
+}
+
+#pragma mark - GADMAdNetworkAdapter
+
++ (nullable Class<GADAdNetworkExtras>)networkExtrasClass {
+  return Nil;
+}
+
++ (NSString *)adapterVersion {
+  return GADMAdapterYahooVersion;
+}
+
+- (id)initWithGADMAdNetworkConnector:(id<GADMAdNetworkConnector>)connector {
+  self = [super init];
+  if (self) {
+    _connector = connector;
+
+    NSDictionary<NSString *, id> *credentials = [connector credentials];
+    if (credentials[GADMAdapterYahooPosition]) {
+      self.placementID = credentials[GADMAdapterYahooPosition];
+    }
+    NSString *siteID = credentials[GADMAdapterYahooDCN];
+    GADMAdapterYahooInitializeYASAdsWithSiteID(siteID);
+  }
+
+  return self;
+}
+
+- (void)dealloc {
+  [self stopBeingDelegate];
+}
+
+#pragma mark - Banner
+
+- (void)getBannerWithSize:(GADAdSize)gadSize {
```

Comment:
I verified banner/inline with our test ads, the banners display and all callbacks are working correctly.